x-pack/plugins/observability_solution/apm/public/components/app/transaction_details/distribution/index.tsx

```diff
@@ -92,6 +92,20 @@ export function TransactionDistribution({
     [history]
   );
 
+  const onSampleClick = useCallback(
+    (sample: any) => {
+      history.push({
+        ...history.location,
+        search: fromQuery({
+          ...toQuery(history.location.search),
+          transactionId: sample.transactionId,
+          traceId: sample.traceId,
+        }),
+      });
```

Comment:
I think we have a `push` method somewhere that is loosely typed (I think it uses some of the old query params). It would be even better if we have a typed `push` method on the router. An example of that is https://github.com/elastic/kibana/blob/72a377d5b2927d75537838b7077e7bd1e340a20f/x-pack/plugins/observability_solution/profiling/public/hooks/use_profiling_router.ts#L29-L30. Obviously out of scope for this PR, but perhaps something that you can use in the future.